app/src/main/java/org/schabi/newpipe/settings/preferencesearch/PreferenceFuzzySearchFunction.java

```diff
@@ -72,39 +72,22 @@ static class FuzzySearchSpecificDTO {
         );
 
         private final PreferenceSearchItem item;
-        private final float score;
+        private final double score;
 
-        FuzzySearchSpecificDTO(
-                final PreferenceSearchItem item,
-                final String keyword) {
+        FuzzySearchSpecificDTO(final PreferenceSearchItem item, final String keyword) {
             this.item = item;
-
-            float attributeScoreSum = 0;
-            int countOfAttributesWithScore = 0;
-            for (final Map.Entry<Function<PreferenceSearchItem, String>, Float> we
-                    : WEIGHT_MAP.entrySet()) {
-                final String valueToProcess = we.getKey().apply(item);
-                if (valueToProcess.isEmpty()) {
-                    continue;
-                }
-
-                attributeScoreSum +=
-                        FUZZY_SCORE.fuzzyScore(valueToProcess, keyword) * we.getValue();
-                countOfAttributesWithScore++;
-            }
-
-            if (countOfAttributesWithScore != 0) {
-                this.score = attributeScoreSum / countOfAttributesWithScore;
-            } else {
-                this.score = 0;
-            }
+            this.score = WEIGHT_MAP.entrySet().stream().mapToDouble(entry -> {
+                final String valueToProcess = entry.getKey().apply(item);
+                return valueToProcess.isEmpty()
+                        ? -1 : FUZZY_SCORE.fuzzyScore(valueToProcess, keyword) * entry.getValue();
+            }).filter(value -> value != -1).average().orElse(0);
```

Comment:
Wouldn't this be better? Also keep the formatting
```suggestion
            this.score = WEIGHT_MAP.entrySet().stream()
                    .map(entry -> new Pair<>(entry.getKey().apply(item), entry.getValue()))
                    .filter(pair -> !pair.first.isEmpty())
                    .mapToDouble(pair -> FUZZY_SCORE.fuzzyScore(pair.first, keyword) * pair.second)
                    .average()
                    .orElse(0);
```